src/Microsoft.ML.Data/Transforms/SelectColumnsTransform.cs

```diff
@@ -0,0 +1,632 @@
+        /// <param name="env">Instance of the host environment</param>
+        /// <param name="columns">The array of column names to keep</param>
+        public SelectColumnsEstimator(IHostEnvironment env, params string[] columns)
+            : this(env, true, (string name) => columns.Contains(name))
+        { }
+
+        /// <summary>
+        /// Constructs the Select Columns Estimator with an array of column names to keep.
+        /// </summary>
+        /// <param name="env">Instance of the host environment</param>
+        /// <param name="keepHidden">Specifies if hidden columns should be removed. Default is true to keep hidden columns</param>
+        /// <param name="columns">The array of column names to keep</param>
+        public SelectColumnsEstimator(IHostEnvironment env, bool keepHidden, params string[] columns)
+            : this(env, keepHidden, (string name) => columns.Contains(name))
+        { }
+
+        /// <summary>
+        /// Constructs the Select Columns Estimator using a specified predicate to determine the columns to keep.
+        /// </summary>
+        /// <param name="env">Instance of the host environment</param>
+        /// <param name="selectPredicate">The predicate that will determines the columns to keep.</param>
+        public SelectColumnsEstimator(IHostEnvironment env, Func<string, bool> selectPredicate)
+            : this(env, true, selectPredicate)
+        { }
+
+        /// <summary>
+        /// Constructs the Select Columns Estimator using a specified predicate to determine the columns to keep.
+        /// </summary>
+        /// <param name="env">Instance of the host environment</param>
+        /// <param name="keepHidden">Specifies if hidden columns should be removed. Default is true to keep hidden columns</param>
+        /// <param name="selectPredicate">The predicate that will determines the columns to keep.</param>
+        public SelectColumnsEstimator(IHostEnvironment env, bool keepHidden, Func<string, bool> selectPredicate)
+        {
+            _host = Contracts.CheckRef(env, nameof(env)).Register(nameof(SelectColumnsEstimator));
+            _selectPredicate = selectPredicate;
+            _keepHidden = keepHidden;
+        }
+
+        public SelectColumnsTransform Fit(IDataView input)
+        {
+            // Generate the list of inputs to select
+            var selectColumns = new HashSet<string>();
+            for (int colIdx = 0; colIdx < input.Schema.ColumnCount; ++colIdx)
+            {
+                var columnName = input.Schema.GetColumnName(colIdx);
+                if (_selectPredicate(columnName))
+                {
+                    selectColumns.Add(columnName);
+                }
+            }
+
+            return new SelectColumnsTransform(_host, _keepHidden, selectColumns.ToArray());
+        }
+
+        public SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            var columns = new List<SchemaShape.Column>();
```

Comment:
how about 
```
var columns = inputSchema.Columns.Where(c=>_selectPredicate(c.Name))
return new SchemaShape(columns)
```

Reply:
So I really want an IEnumerable for Columns, but all I have is ColumnCount and GetColumnName(idx). Maybe this is coming in the near future? Oh wait -- I just saw that your PR went though. I will update...

---
In reply to: [225387439](https://github.com/dotnet/machinelearning/pull/1269#discussion_r225387439) [](ancestors = 225387439)

Reply:
Ok thank you - I have updated.

---
In reply to: [225682139](https://github.com/dotnet/machinelearning/pull/1269#discussion_r225682139) [](ancestors = 225682139,225387439)